src/firmware/linux/host/types/snp.rs

```diff
@@ -249,6 +249,71 @@ impl Default for SnpSetConfig {
     }
 }
 
+// Length defined in the Linux Kernel for the IOCTL.
+const VLEK_BUFFER_LEN: usize = 432;
+
+#[cfg(feature = "snp")]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[repr(C, packed)]
+/// Wrapped VLEK data.
+pub struct WrappedVlekHashstick<'a> {
+    /// Opaque data provided by AMD Key Distribution Server
+    /// (as described in SEV-SNP Firmware ABI 1.54, SNP_VLEK_LOAD)
+    pub data: &'a [u8], // 432 bytes of data
+}
+
+impl<'a, 'b: 'a> std::convert::TryFrom<&'b [u8]> for WrappedVlekHashstick<'a> {
+    type Error = VlekError;
+
+    fn try_from(value: &'b [u8]) -> Result<Self, Self::Error> {
+        if value.len() != VLEK_BUFFER_LEN {
+            return Err(VlekError::InvalidLength);
+        }
+
+        if value == [0u8; VLEK_BUFFER_LEN] {
+            return Err(VlekError::EmptyCertBuffer);
+        }
+
+        Ok(Self { data: value })
+    }
+}
+
+#[cfg(feature = "snp")]
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq)]
+#[repr(C, packed)]
+/// Structure used to load a VLEK hashstick into the AMD Secure Processor.
+pub struct SnpVlekLoad {
+    /// Length of the command buffer read by the AMD Secure Processor.
+    pub len: u32,
+
+    /// Version of wrapped VLEK hashstick (Must be 0h).
+    pub vlek_wrapped_version: u8,
+
+    _reserved: [u8; 3],
+
+    /// Address of wrapped VLEK hashstick ([WrappedVlekHashstick])
+    pub vlek_wrapped_address: u64,
+}
+
+#[cfg(feature = "snp")]
+impl SnpVlekLoad {
+    /// Creates a new VLEK load instruction from a hashstick.
+    pub fn new(hashstick: &WrappedVlekHashstick) -> Self {
+        hashstick.into()
+    }
+}
+
```

Comment:
So the use scenario for a user is that they provide the raw bytes into a SNPVlekLoad struct?

Reply:
In this case, the user will provide the hashstick blob to the API, and the API will wrap the contents into a `WrappedVlekHashstick`, then that will be wrapped into the `SnpVlekLoad` command structure, which is subsequently passed to the IOCTL. Once the hashstick is loaded into the ASP, the platform owner will provide the certificate to the guest via the extended guest request mechanism.